### DISCUSSION TITLE
ValueError: Digitalized labels does not have elements close enough to bin index 4. The bin index should be in the range of the labels values.

### DISCUSSION THREAD

**Discussion by zhuyuhang4** [AdaptiveMotorControlLab/CEBRA#125] [Q&A] [answered]
![cerbra](https://github.com/AdaptiveMotorControlLab/CEBRA/assets/144974755/5d56ad61-be4c-4821-86f1-bbc6e5f90eaa)
I see the example program labels are continuous data, but my labels are discrete values 0, 1, 2, and I get an error. I seem to make it work by changing n_bins.

**Comment by stes** [comment, marked as answer]
Hi @zhuyuhang4 , this is expected behavior --- you need a sufficient number of samples within each bin of the histogram that is internally created. Do you have some example code to repro your issue? But effectively, using a smaller number of bins is the correct way to go.

**Comment by stes** [comment]
I created an issue to improve the code here: https://github.com/AdaptiveMotorControlLab/CEBRA/issues/128

In the meantime, please consider setting n_bins manually to the number of discrete labels. Let me know if this works!

**Reply by MMathisLab** [reply]
I think a short demo on how to set the bins would be useful here and in the issue

**Comment by stes** [comment]
Hi @zhuyuhang4 , did this address your question/were you able to solve the issue? If so, please mark the answer, or happy to address any follow up question!
